src/settings.rs

```diff
@@ -358,8 +345,13 @@ fn settings_from(matches: &ArgMatches) -> (Settings, RunMode) {
 }
 
 /// Get environment variable and turn it into integer.
-fn get_env_var_value(name: &str) -> Result<u8, std::env::VarError> {
-    env::var(name).map(|value| u8::from_str(&value).unwrap_or(1))
+fn get_env_var_value(name: &str) -> Option<u8> {
+    env::var_os(name).filter(|v| !v.is_empty()).map(|value| {
+        value
+            .to_str()
+            .and_then(|v| v.parse::<u8>().ok())
+            .unwrap_or(1)
+    })
```

Comment:
```suggestion
    let value_str = env::var_os(name).filter(|v| !v.is_empty())?;
    let value = value_str
        .to_str()
        .and_then(|v| v.parse::<u8>().ok())
        .unwrap_or(1);
    Some(value)
```

Reply:
Given there's 2 layers of Option here, I think I'd like to keep the logic as it is - looking at the suggested version my instinct was to change it to `.to_str()?.parse().unwrap_or(1)`, but that's different. I think the explicit `map()` makes more clear the control flow and conditions.

Reply:
line 349 and the following lines are unrelated logics. early return reveals line 350 is independent to 349 and code readers can forget about 349 when going to 350.
Using `map` make readers essentially follow the full logic without benefit.